File: python/program/PetShop.py
# PetShop.py
from Dog import Dog
from Cat import Cat
from Bird import Bird
from PetProduct import PetProduct

class PetShop:
    def __init__(self, name="", location="", dogs=None, cats=None, birds=None, products=None):
        self.__name = name
        self.__location = location
        self.__dogs = dogs if dogs is not None else []
        self.__cats = cats if cats is not None else []
        self.__birds = birds if birds is not None else []
        self.__products = products if products is not None else []
    
    def set_name(self, name):
        self.__name = name
    
    def set_location(self, location):
        self.__location = location
    
    def set_dogs(self, dogs):
        self.__dogs = dogs
    
    def set_cats(self, cats):
        self.__cats = cats
    
    def set_birds(self, birds):
        self.__birds = birds
    
    def set_products(self, products):
        self.__products = products
    
    def add_dog(self, dog):
        self.__dogs.append(dog)
    
    def add_cat(self, cat):
        self.__cats.append(cat)
    
    def add_bird(self, bird):
        self.__birds.append(bird)
    
    def add_product(self, product):
        self.__products.append(product)
    
    def get_name(self):
        return self.__name
    
    def get_location(self):
        return self.__location
    
    def get_dogs(self):
        return self.__dogs
    
    def get_cats(self):
        return self.__cats
    
    def get_birds(self):
        return self.__birds
    
    def get_products(self):
        return self.__products

File: java/program/PetShop.java

import java.util.ArrayList;
import java.util.List;

public class PetShop {
    private String name;
    private String location;
    private List<Dog> dogs;
    private List<Cat> cats;
    private List<Bird> birds;
    private List<PetProduct> products;
    
    public PetShop() {
        this.dogs = new ArrayList<>();
        this.cats = new ArrayList<>();
        this.birds = new ArrayList<>();
        this.products = new ArrayList<>();
    }
    
    public PetShop(String name, String location, List<Dog> dogs, List<Cat> cats, List<Bird> birds, List<PetProduct> products) {
        this.name = name;
        this.location = location;
        this.dogs = dogs;
        this.cats = cats;
        this.birds = birds;
        this.products = products;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public void setLocation(String location) {
        this.location = location;
    }
    
    public void setDogs(List<Dog> dogs) {
        this.dogs = dogs;
    }
    
    public void setCats(List<Cat> cats) {
        this.cats = cats;
    }
    
    public void setBirds(List<Bird> birds) {
        this.birds = birds;
    }
    
    public void setProducts(List<PetProduct> products) {
        this.products = products;
    }
    
    public void addDog(Dog dog) {
        this.dogs.add(dog);
    }
    
    public void addCat(Cat cat) {
        this.cats.add(cat);
    }
    
    public void addBird(Bird bird) {
        this.birds.add(bird);
    }
    
    public void addProduct(PetProduct product) {
        this.products.add(product);
    }
    
    public String getName() {
        return this.name;
    }
    
    public String getLocation() {
        return this.location;
    }
    
    public List<Dog> getDogs() {
        return this.dogs;
    }
    
    public List<Cat> getCats() {
        return this.cats;
    }
    
    public List<Bird> getBirds() {
        return this.birds;
    }
    
    public List<PetProduct> getProducts() {
        return this.products;
    }
}